### PR TITLE
[Color system] Update names

### PR DESCRIPTION
**Pull request by tmlayton**
### WHY are these changes introduced?

Updates color names per our working sessions in preparation for user testing.

### WHAT is this pull request doing?

Updates names per [these changes](https://github.com/Shopify/polaris-react/compare/color-system-doc-testing?short_path=7e44770#diff-7e44770a23ffedfb5ca5fa1d9856024b)

## 

### How to 🎩

- Poke around the kitchen sink in light and dark modes to make sure things look okay

* [x] 🎩

### CODE REVIEW

File: src/components/Avatar/Avatar.scss [hunk 1]
```diff
@@ -77,7 +77,7 @@ $large-size: rem(60px);
   left: 50%;
   width: 100%;
   height: 100%;
-  background-color: var(--p-neutral, color('sky'));
+  background-color: var(--p-secondary, color('sky'));
```
Comment (danrosenthal) [inline src/components/Avatar/Avatar.scss]: We should use this pull request as an opportunity to get rid of the base (unmodified) variants. They've been more confusing than useful.

Update this to use the disabled secondary action like `Badge` uses.

```suggestion
background-color: var(--p-action-secondary-disabled, color('sky'));
```

File: src/utilities/theme/tokens.ts [hunk 2]
```diff
@@ -21,13 +21,13 @@ const Overrides = {
...
-  bannerWarningBorder: buildBannerBorder('--p-warning-border'),
-  bannerCriticalBorder: buildBannerBorder('--p-critical-border'),
+  bannerBorderDefault: buildBannerBorder('--p-border'),
+  bannerBorderSuccess: buildBannerBorder('--p-border-success'),
+  bannerBorderHighlight: buildBannerBorder('--p-border-highlight'),
+  bannerBorderWarning: buildBannerBorder('--p-border-warning'),
+  bannerBorderCritical: buildBannerBorder('--p-border-critical'),
   badgeMixBlendMode: 'luminosity',
-  borderSubdued: `${rem('1px')} solid var(--p-border-subdued-on-surface)`,
+  thinBorderSubdued: `${rem('1px')} solid var(--p-border-subdued)`,
```
Comment (tmlayton) [inline src/utilities/theme/tokens.ts]: This one is a little weird but there was a naming conflict

File: src/components/Avatar/Avatar.scss (hunk 1 above)
Comment (tmlayton) [reply src/components/Avatar/Avatar.scss]: Okay, double checked this was the only one

**Comment by tmlayton** [comment]
> Looks the order of the naming for the `primary`, `critical`, `warning`, `highlight`, and `success` buckets needs to be updated to element first (`action-primary`, `border-critical`).
> 
> The other buckets look good 👍

Yep, was still WIP. Finished now and will go ahead and merge so we can move forward with the user testing sandbox (I’d rather have the updated color docs in master). Please still review for possible missed names and fix forward.

**Pull request merged**